### PR TITLE
PMM-9320 Unable to add MongoDB service due to + in the password

### PR DESCRIPTION
**Pull request by mihailo-chehivsky**
PMM-9320

### CODE REVIEW

File: models/agent_model.go [hunk 1]
```diff
@@ -436,15 +436,18 @@ func (s *Agent) DSN(service *Service, dialTimeout time.Duration, database string
 			Path:     path,
 			RawQuery: q.Encode(),
 		}
+		plusPlaceholder := "__plus__" // There is a risk of error, when the password contains this string. Can change it to a more unique string.
```
Comment (ritbl) [inline models/agent_model.go:439]: why not change to a constant?

**Comment by ritbl** [comment]
@mihailo-chehivsky in general I don't like this workaround. is possible to escape the password?
if not I'd rather consider using `hash(hash(pwd)+salt)` instead of a password. How have you tested it?

**Review by ritbl — CHANGES_REQUESTED** [review]
see my comments

**Comment by ritbl** [comment]
@mihailo-chehivsky it is not an escape, what about other potential issues (with other characters). What about my suggestion?

**Comment by mihailo-chehivsky** [comment]
> @mihailo-chehivsky it is not an escape, what about other potential issues (with other characters). What about my suggestion?
> 
> > Tested adding a new instance MongoDB using a password contains "+" on development environment.
> 
> so you run devcontainer? did you test via pmm-admin or via UI?

vi UI

**Comment by mihailo-chehivsky** [comment]
@ritbl 
I guess this bug relates to "+" character only. 
Similar issue described here: https://stackoverflow.com/questions/2678551/when-should-space-be-encoded-to-plus-or-20

**Pull request closed without merging**